azerothcore/azerothcore-wotlk
# fix(DB/creature_loot): Remove invalid Purple Lotus drops from NPCs

## Changes Proposed:
- Removes invalid [Purple Lotus](https://tbc.wowhead.com/item=8831/purple-lotus#dropped-by;0-2-10+9+1) from 285 NPCs who drop it incorrectly. Wowhead shows it's exclusively from various plant elementals, but on AC it's dropped by lots of unrelated NPCs, usually with a 0.02% rate.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8372
- Closes https://github.com/chromiecraft/chromiecraft/issues/1851

## SOURCE:

https://tbc.wowhead.com/item=8831/purple-lotus#dropped-by;0+9-10+1
https://wowpedia.fandom.com/wiki/Purple_Lotus?oldid=2347781

## Tests Performed:

- Ran SQL, no errors/warnings, 285 rows changed as expected.

## How to Test the Changes:


Check using this. It should now return no results.
```SQL
SELECT ct.entry, ct.name, ct.maxlevel, ct.rank, clt.Chance
FROM `creature_loot_template` clt
JOIN `creature_template` ct ON ct.lootid = clt.Entry
WHERE clt.Item = 8831
AND ct.entry NOT IN (12219, 6510, 6509, 6511, 6512, 12223, 12224, 12220, 5881, 8384, 13141, 
13142, 5490, 7139, 14448, 1812, 6517, 6527, 6519, 6518, 5481, 5485, 7584, 7100, 7101, 11462, 
13022)
```

## Known Issues and TODO List:


N/A.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
